### PR TITLE
Added info about the prompts

### PR DESCRIPTION
**Pull request by mja00**
I spent a day testing the prompt features and figured out what type of params are needed for all of these to work. I also created comments on the natives to explain what some settings change.

### CODE REVIEW

File: natives.json [hunk 1]
```diff
@@ -28277,18 +28277,18 @@
...
 			"params": [
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "Prompt",
+					"name": "prompt"
 				},
 				{
-					"type": "Any",
+					"type": "Integer",
```
Comment (UnknownModder) [inline natives.json]: This is actually a ``float``.

File: natives.json [hunk 2]
```diff
@@ -28277,18 +28277,18 @@
...
 					"name": "p1"
 				},
 				{
-					"type": "Any",
+					"type": "Boolean",
```
Comment (UnknownModder) [inline natives.json]: And this should be an ``int`` (name: ``direction``).

Comment (mja00) [reply natives.json]: I can promise it's not an int because when used ingame the int did nothing.

File: natives.json (hunk 1 above)
Comment (mja00) [reply natives.json]: It's actually not a float.

File: natives.json (hunk 2 above)
Comment (mja00) [reply natives.json]: I'll rephrase that setting it to an int anywhere from 1 to 10 resulted in the prompt asking for a CCW rotation. When it's a bool it then asks for CCW or CW. Setting this param to 0 or anything higher than 10 causes the native to fail. So please tell me what it should be again. :)

**Comment by mja00** [comment]
I've updated the types to C-style, I changed nothing else because I believe those types to be the best representation of what is actually accepted by the game.

File: natives.json (hunk 1 above)
Comment (UnknownModder) [reply natives.json]: Yes it is.
```cpp
void __fastcall _PROMPT_SET_ROTATE_MODE(int promptHandle, float a2, int direction)
{
  int _direction; // ebx
  __int64 prompt; // rax

  _direction = direction;
  prompt = CGameScriptHandler::GetScriptPrompt(promptHandle);
  if ( prompt )
    Prompt::SetRotateMode(prompt, a2, _direction);
}
```
```cpp
void __fastcall Prompt::SetRotateMode(__int64 this, float a2, int direction)
{
  __int64 _this; // rbx
  int _direction; // edi
  __int64 v5; // rax
  __int64 v6; // rcx

  _this = this;
  _direction = direction;
  v5 = operator new(0x30ui64);
  if ( v5 )
  {
    *(_QWORD *)v5 = &PromptMode::`vftable';
    v6 = (unsigned int)TlsIndex;
    *(_QWORD *)(v5 + 8) = _this + 0x30;
    if ( !*(_QWORD *)(*(_QWORD *)(__readgsqword(0x58u) + 8 * v6) + 0x12B8i64) )
      *(_WORD *)(v5 + 0x10) = 0;
    *(_BYTE *)(v5 + 0x18) &= ~3u;
    *(_QWORD *)v5 = &RotatePromptMode::`vftable';
    *(float *)(v5 + 0x24) = a2;
    *(_DWORD *)(v5 + 0x14) = 0;
    *(_DWORD *)(v5 + 0x20) = 0;
    *(_DWORD *)(v5 + 0x28) = 0;
    *(_DWORD *)(v5 + 0x2C) = _direction;
  }
  else
  {
    v5 = 0i64;
  }
  sub_140317E34(_this, v5);
}
```
```asm
movss   dword ptr [rax+24h], xmm6
```

File: natives.json (hunk 2 above)
Comment (UnknownModder) [reply natives.json]: And again, it is definitely an int, see my comment above.

**Comment by UnknownModder** [comment]
So are you gonna fix those two params or not?
If not, then I'll just close this tomorrow and fix it myself in my next commit.

**Pull request closed without merging**